src/mono/wasi/Wasi.Build.Tests/PInvokeTableGeneratorTests.cs

```diff
@@ -0,0 +1,76 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO;
+using Xunit;
+using Xunit.Abstractions;
+using Wasm.Build.Tests;
+
+#nullable enable
+
+namespace Wasi.Build.Tests;
+
+public class PInvokeTableGeneratorTests : BuildTestBase
+{
+    public PInvokeTableGeneratorTests(ITestOutputHelper output, SharedBuildPerTestClassFixture buildContext)
+        : base(output, buildContext)
+    {
+    }
+
+    [Fact]
+    public void InteropSupportForUnmanagedEntryPointWithoutDelegate()
+    {
+        string config = "Release";
+        string id = $"{config}_{GetRandomId()}";
+        string projectFile = CreateWasmTemplateProject(id, "wasiconsole");
+        string code =
+                """
+                using System;
+                using System.Runtime.InteropServices;
+                public unsafe class Test
+                {
+                    [UnmanagedCallersOnly(EntryPoint = "ManagedFunc")]
+                    public static int MyExport(int number)
+                    {
+                        // called from MyImport aka UnmanagedFunc
+                        Console.WriteLine($"MyExport({number}) -> 42");
+                        return 42;
+                    }
+
+                    [DllImport("*", EntryPoint = "UnmanagedFunc")]
+                    public static extern void MyImport(); // calls ManagedFunc aka MyExport
+
+                    public unsafe static int Main(string[] args)
+                    {
+                        Console.WriteLine($"main: {args.Length}");
+                        MyImport();
+                        return 0;
+                    }
+                }
+                """;
+
+        string projectName = Path.GetFileNameWithoutExtension(projectFile);
+
+        var buildArgs = new BuildArgs(projectName, config, AOT: true, ProjectFileContents: id, ExtraBuildArgs: null);
+        buildArgs = ExpandBuildArgs(buildArgs);
+        AddItemsPropertiesToProject("<NativeFileReference Include=\"local.c\" />");
+        AddItemsPropertiesToProject(projectFile, "<WasmSingleFileBundle>true</WasmSingleFileBundle>");
+
+        BuildProject(buildArgs,
+                    id: id,
+                    new BuildProjectOptions(
+                        InitProject: () =>
+                        {
+                            File.WriteAllText(Path.Combine(_projectDir!, "Program.cs"), code);
```

Comment:
Not sure why, but this line doesn't overwrite Program.cs.

Reply:
It's execution is guarded by `options.CreateProject` https://github.com/dotnet/runtime/blob/71e989d1b5641024655800cbae43f62895544692/src/mono/wasi/Wasi.Build.Tests/BuildTestBase.cs#L208

It's value should be true by default, so it should be executed. Anyway I think we don't want `CreateProject=true`, because the project is already created by `CreateWasmTemplateProject`. So I would just override `Program.cs` before running `BuildProject`